include/flucoma/clients/nrt/DataSetClient.hpp

```diff
@@ -413,13 +413,12 @@ class DataSetRead : public FluidBaseClient, ControlIn, ControlOut
                 // unavailable");
 
       auto outputDSpointer = get<kOutputDataSet>().get().lock();
-      if (!outputDSpointer)
-        return; // c.reportError("FluidDataSet RT Query could not obtain
-                // reference (output) FluidDataSet");  //TODO: we might as well
-                // consider passing the distances in this case (and in kdtree
-                // too!)
+      //      if (!outputDSpointer)
+      //        return; // c.reportError("FluidDataSet RT Query could not obtain
+      // reference (output) FluidDataSet");
 
-      index pointSize = outputDSpointer->dims();
+      index pointSize =
+          outputDSpointer ? (index) outputDSpointer->dims() : (index) 1;
```

Comment:
No need for casts here. `dims()` is actually returning a `MessageResult<index>` (for consumption by the host), just call `value()` on that:
```suggestion
         outputDSpointer ? outputDSpointer->dims().value() : 1;
```
(And when we do cast in C++, bear in mind that C-style casts are rarely the thing you want: here `static_cast<T>` would have been the thing)